File: src/main/java/org/example/inventorysystem/models/Asset.java
package org.example.inventorysystem.models;

import com.fasterxml.jackson.annotation.*;
import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "asset")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Asset {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "person_id")
	private Person person;

	private Integer inventoryNumber;
	private String name;
	private Long value;
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy")
	private LocalDate date;

	@Nullable
	private String adnotations;
	private AssetStatus status;

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "room_id")
	private Room room;

	@ManyToOne
	@JsonBackReference("inventoryField-assets")
	private InventoryField inventoryField;

	@Nullable
	private Type type;
}

File: front/inventory_app/src/app/common/models/asset.ts
import {Person} from './person';
import {
  AssetStatus
} from './asset_status';
import {Room} from './room';
import {
  InventoryField
} from './inventory-field';
import {TypeEnum} from '../enumes/typeEnum';

export interface Asset {
  id?: number | null;
  person?: Person;
  inventoryNumber?: number | null;
  name?: string;
  value?: number;
  date?: string | number | Date;
  adnotations?: string;
  status?: AssetStatus;
  room?: Room;
  inventoryField?: InventoryField;
  type?: TypeEnum;
}

